prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/git/GitBranchTaskTest.java

```diff
@@ -0,0 +1,162 @@
+package com.redhat.parodos.tasks.git;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.utils.WorkContextUtils;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Slf4j
+class GitBranchTaskTest {
+
+	private GitBranchTask gitBranchTask;
+
+	private Path gitRepoPath;
+
+	private Repository repository;
+
+	private Path tempDir;
+
+	@BeforeEach
+	public void setUp() throws Exception {
+		gitBranchTask = new GitBranchTask();
+		gitBranchTask.setBeanName("GitBranchTask");
+		tempDir = Files.createTempDirectory("git-repo");
+
+		gitRepoPath = tempDir.resolve(".git");
+
+		this.repository = FileRepositoryBuilder.create(gitRepoPath.toFile());
+		this.repository.create();
+		log.info("Created a new repository at '{}'", this.repository.getDirectory());
+		this.createSingleFileInRepo();
+	}
+
+	@AfterEach
+	public void tearDown() throws Exception {
+		assertDoesNotThrow(() -> {
+			repository.close();
+		});
+		try (Stream<Path> walk = Files.walk(tempDir)) {
+			walk.sorted(java.util.Comparator.reverseOrder()).map(Path::toFile).forEach(p -> {
+				assertDoesNotThrow(() -> Files.delete(p.toPath()));
+			});
+		}
+	}
+
+	@Test
+	public void testBranchCheckout() {
+		// given
+		createSingleFileInRepo();
+		WorkContext workContext = new WorkContext();
+		workContext.put("path", tempDir.toString());
+		WorkContextUtils.addParameter(workContext, "branch", "newBranch");
+		WorkContextUtils.addParameter(workContext, "path", tempDir.toString());
+
+		// when
+		var result = this.gitBranchTask.execute(workContext);
+
+		// then
+		assertNull(result.getError());
+		assertEquals(result.getStatus(), WorkStatus.COMPLETED);
+
+		assertDoesNotThrow(() -> {
+			Ref branchRef = repository.findRef("newBranch");
+			assertNotNull(branchRef);
+			assertEquals(repository.getBranch(), "newBranch");
+		});
+	}
+
+	@Test
+	public void testBranchWithAlreadyBranch() {
+		// given
+		createSingleFileInRepo();
+
+		assertDoesNotThrow(() -> {
+			assertEquals(repository.getBranch(), "master");
```

Comment:
Please change "master" to "main" and since it's repeated, could be best to user a static variable